addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -19,62 +19,92 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
         var $amountUntaxed = $('#order_total_untaxed span.oe_currency_value');
         var $amountTax = $('#order_total_taxes span.oe_currency_value');
         var $amountTotal = $('#order_total span.oe_currency_value');
-        var $carrierBadge = $('#delivery_carrier input[name="delivery_type"][value=' + result.carrier_id + '] ~ .badge.d-none');
-        var $computeBadge = $('#delivery_carrier input[name="delivery_type"][value=' + result.carrier_id + '] ~ .o_delivery_compute');
-        var $discount = $('#order_discounted');
-
-        if ($discount && result.new_amount_order_discounted) {
-            // Cross module without bridge
-            // Update discount of the order
-            $discount.find('.oe_currency_value').text(result.new_amount_order_discounted);
-
-            // We are in freeshipping, so every carrier is Free
-            $('#delivery_carrier .badge').text(_t('Free'));
-        }
 
         if (result.status === true) {
             $amountDelivery.text(result.new_amount_delivery);
             $amountUntaxed.text(result.new_amount_untaxed);
             $amountTax.text(result.new_amount_tax);
             $amountTotal.text(result.new_amount_total);
-            $carrierBadge.children('span').text(result.new_amount_delivery);
-            $carrierBadge.removeClass('d-none');
-            $computeBadge.addClass('d-none');
             $payButton.data('disabled_reasons').carrier_selection = false;
             $payButton.prop('disabled', _.contains($payButton.data('disabled_reasons'), true));
         } else {
-            console.error(result.error_message);
-            $computeBadge.text(result.error_message);
             $amountDelivery.text(result.new_amount_delivery);
             $amountUntaxed.text(result.new_amount_untaxed);
             $amountTax.text(result.new_amount_tax);
             $amountTotal.text(result.new_amount_total);
         }
     },
+    /**
+     * @private
+     * @param {Object} result
+     */
+    _handleCarrierUpdateResultBadge: function (result) {
+        var $carrierBadge = $('#delivery_carrier input[name="delivery_type"][value=' + result.carrier_id + '] ~ .badge.d-none');
+        var $computeBadge = $('#delivery_carrier input[name="delivery_type"][value=' + result.carrier_id + '] ~ .o_delivery_compute');
+
+        if (result.status === true) {
+             // if free delivery (`free_over` field), show 'Free', not '$0'
+             if (parseInt(result.new_amount_delivery)) {
+                 $carrierBadge.children('span').text(result.new_amount_delivery);
+             } else {
+                 $carrierBadge.text(_t('Free'));
+             }
+             $carrierBadge.children('span').text(result.new_amount_delivery);
+             $carrierBadge.removeClass('d-none');
+             $computeBadge.addClass('d-none');
+             $computeBadge.removeClass('o_wsale_delivery_carrier_error');
+        } else {
+            console.error(result.error_message);
```

Comment:
can we remove this ?

Reply:
I don't know, I prefer not do, it does no harm and it was part of the IMP commit https://github.com/odoo/odoo/commit/da3c561d35452bb45921bbaa492eb4d9b36b0233#diff-54ba6dd06746c8beb778b0150f0e37f7R26 that was reviewed so I don't think it should be removed

Reply:
I mean it is not an console.error that was forgotten during the development phase.
It was explicitly and willingly added there.
And as it does no harm, that was not the purpose of this PR to remove that code.

> My idea is, if one day with have a crash manager. Do we need to show this error in a modal like a traceback?

One day..  :notes:
I don't know.. Was I supposed to decided on this PR what should be done for the possible futur task of the crash manager?

> So If I follow your idea, we cannot make any fix and improvement since the initial code was reviewed previously?

Removing that console.error would neither be a fix or an improvement that's why I did not touch that line (that does no harm).

But I am not against removing that line now that it has been debated and that I also think that it will be better to not throw an error for the Odoo v18 when we will have the crash manager
I will remove it

Reply:
![](https://media.giphy.com/media/l46CsKjjCKkZGVfzi/giphy.gif)

Reply:
ok thank you I will remove it myself  before Odoo v18.

Reply:
And yes, it is a improvement to avoid to log error while it is a catched and expected error.

Reply:
thanks for your help, we will finalize this task internally